auth0_flutter/lib/src/web/auth0_flutter_plugin_real.dart

```diff
@@ -1,13 +1,41 @@
-import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart';
+import 'dart:html';
+import 'dart:js_util';
+
+import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart'
+    show Account, Auth0FlutterWebPlatform, LoginOptions;
+
 import 'package:flutter_web_plugins/flutter_web_plugins.dart';
 
-// import 'js_interop.dart';
+import 'js_interop.dart';
 
 class Auth0FlutterPlugin extends Auth0FlutterWebPlatform {
   static void registerWith(final Registrar registrar) {
     Auth0FlutterWebPlatform.instance = Auth0FlutterPlugin();
   }
 
-  // methods, e.g. loginWithRedirect()
-  // here they should call the interop methods, that in turn call the JS methods
+  late Auth0Client client;
+
+  @override
+  Future<void> initialize(final Account account,
+      {final AuthorizationParams? authorizationParams}) async {
+    client = Auth0Client(
```

Comment:
Should we create a new instance only when there isn' one already?

Reply:
The more I play with this, the more I don't think it's possible or at least intuitive.

I can do:

```dart
if (client == null) {
  client = Auth0Client(Auth0ClientOptions(
      domain: account.domain, clientId: account.clientId));
}
```

But because `client` is non-optional, the block never runs - I guess the compiler is optimizing out the statement as it knows it'll never be executed. Same if I try:

```dart
client ??= Auth0Client(Auth0ClientOptions(
      domain: account.domain, clientId: account.clientId));
```

And actually the linter warns me about this one.

We could track some internal state though to tell whether the client has been initialized. We could also type it as `Auth0Client?` and manage possible null references everywhere, but that's a bit more of a pain.

Reply:
Let's leave it this way then.